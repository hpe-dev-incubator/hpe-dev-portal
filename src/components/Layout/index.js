import React from 'react';
import PropTypes from 'prop-types';
import { Grommet, Box, Main } from 'grommet';
import { useDarkMode } from '../../hooks';

import theme from './theme';
import './reset.css';
import { Footer, Header } from '../index';

const Layout = ({ children }) => {
  const { value } = useDarkMode(true);
  const themeMode = value ? 'dark' : 'light';
  return (
    <Grommet theme={theme} themeMode={themeMode}>
      <Box direction="row" justify="center">
        <Box basis="xxlarge" flex="shrink">
          <Header />
          <Main flex={false} fill={undefined} overflow="visible">
            {children}
          </Main>
          <Footer />
        </Box>
      </Box>
    </Grommet>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
