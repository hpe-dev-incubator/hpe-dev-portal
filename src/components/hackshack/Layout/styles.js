import styled from 'styled-components';
import { Box, Layer } from 'grommet';

const ResponsiveLayout = styled(Box)`
  min-width: 366px;
  min-height: 100%;
  @supports (-moz-appearance: none) {
    filter: ${(props) => (props.layer ? 'blur(15px)' : 'none')};
  }
`;

const StyledLayer = styled(Layer)`
  background-color: rgba(38, 48, 64, 0.8);
  backdrop-filter: blur(15px);
`;

export { ResponsiveLayout, StyledLayer };
