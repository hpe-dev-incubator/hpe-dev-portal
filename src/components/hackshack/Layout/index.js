import React, {
  Children,
  cloneElement,
  isValidElement,
  useContext,
  useState,
} from 'react';
import PropTypes from 'prop-types';
import { Grommet, Box, ResponsiveContext, Text, Button } from 'grommet';
import { Close } from 'grommet-icons';
import { hpe } from 'grommet-theme-hpe';
import { deepMerge } from 'grommet/utils';
import { ResponsiveLayout, StyledLayer } from './styles';
import { Header as HackShackHeader, SideNav } from '../index';
import { Header as HPEDevHeader } from '../../index';

const customHpe = deepMerge(hpe, {
  global: {
    breakpoints: {
      small: {
        value: 900,
      },
    },
  },
});

const Layout = ({ children, background }) => {
  const [layer, setLayer] = useState(false);
  const size = useContext(ResponsiveContext);

  const childrenWithProps = Children.map(children, (child) => {
    if (isValidElement(child)) {
      return cloneElement(child, { size });
    }

    return child;
  });

  return (
    <Grommet
      theme={customHpe}
      themeMode="dark"
      background="#151d29"
      style={{ overflowX: 'hidden' }}
    >
      <ResponsiveLayout
        background={{
          image: `url(${background})`,
          size: '100%',
          position: 'top center',
        }}
        justify="between"
        layer={layer}
      >
        <Box>
          <HPEDevHeader />
          <HackShackHeader setLayer={setLayer} size={size} />
          <Box direction="row">
            {size !== 'small' && (
              <Box margin={{ top: 'xlarge', left: 'large' }}>
                <SideNav size={size} />
              </Box>
            )}
            <Box
              align={size !== 'small' ? 'start' : 'center'}
              fill="horizontal"
              direction="column"
              pad="xlarge"
            >
              {childrenWithProps}
            </Box>
          </Box>
        </Box>
        {layer && (
          <StyledLayer>
            <Box pad={{ top: 'xlarge', right: 'large' }}>
              <Box
                direction="row"
                align="center"
                justify="end"
                margin={{ bottom: 'xlarge' }}
              >
                <Text color="#FFFFFF">CLOSE</Text>
                <Button icon={<Close />} onClick={() => setLayer(false)} />
              </Box>
              <Box align="start" gap="large" pad="xlarge">
                <SideNav size={size} />
              </Box>
            </Box>
          </StyledLayer>
        )}
      </ResponsiveLayout>
    </Grommet>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  background: PropTypes.string.isRequired,
};

export default Layout;
