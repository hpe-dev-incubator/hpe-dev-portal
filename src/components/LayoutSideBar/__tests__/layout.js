import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery } from 'gatsby';
import Layout from '../index';

beforeEach(() => {
  useStaticQuery.mockReturnValue({
    site: {
      siteMetadata: {
        title: 'My site',
        author: 'Tester',
      },
    },
  });
});

describe('Layout', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Layout width="medium">This is my app wrapper</Layout>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
