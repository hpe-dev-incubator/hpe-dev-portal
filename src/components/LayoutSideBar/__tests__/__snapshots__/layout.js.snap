// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 edlFAH"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "auto",
      "maxWidth": "100%",
      "minHeight": "100vh",
    }
  }
>
  <div
    className="StyledBox-sc-13pk1d4-0 dmrmMT"
  >
    <span
      className="StyledText-sc-1sadyjn-0 jRUGKF"
    >
      HPE Developer
    </span>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
    />
    <div
      className="StyledBox-sc-13pk1d4-0 exddWn"
    >
      <a
        className="sc-bdVaJa bwmoIK"
        href="/research"
      >
        Research
      </a>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
      />
      <a
        className="sc-bdVaJa bwmoIK"
        href="/design"
      >
        Design
      </a>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
      />
      <a
        className="sc-bdVaJa bwmoIK"
        href="/develop"
      >
        Develop
      </a>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
      />
      <a
        className="sc-bdVaJa bwmoIK"
        href="/support"
      >
        Support
      </a>
    </div>
  </div>
  <main
    className="StyledBox-sc-13pk1d4-0 gYrpnw"
  >
    This is my app wrapper
  </main>
  <div
    className="StyledBox-sc-13pk1d4-0 IRSNj"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 hoKQJh"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 exddWn"
      >
        <a
          className="sc-bdVaJa bwmoIK"
          href="/opensource"
        >
          Open Source
        </a>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
        />
        <a
          className="sc-bdVaJa bwmoIK"
          href="/events"
        >
          Events
        </a>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
        />
        <a
          className="sc-bdVaJa bwmoIK"
          href="/community"
        >
          Community
        </a>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 exddWn"
      >
        <a
          aria-label="Follow us on Twitter"
          className="StyledAnchor-sc-1rp7lwl-0 hdTGql"
          href="https://twitter.com/"
          onBlur={[Function]}
          onFocus={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-label="Twitter"
            className="StyledIcon-ofa7kd-0 fDZBFh"
            viewBox="0 0 24 24"
          >
            <path
              d="M24,4.3086 C23.117,4.7006 22.168,4.9646 21.172,5.0836 C22.188,4.4746 22.969,3.5096 23.337,2.3596 C22.386,2.9246 21.332,3.3336 20.21,3.5556 C19.312,2.5976 18.032,1.9996 16.616,1.9996 C13.897,1.9996 11.692,4.2046 11.692,6.9236 C11.692,7.3096 11.736,7.6856 11.82,8.0456 C7.728,7.8406 4.099,5.8806 1.671,2.9006 C1.247,3.6286 1.004,4.4736 1.004,5.3766 C1.004,7.0846 1.873,8.5926 3.195,9.4756 C2.388,9.4486 1.628,9.2276 0.964,8.8596 L0.964,8.9206 C0.964,11.3066 2.661,13.2966 4.914,13.7486 C4.501,13.8626 4.065,13.9216 3.617,13.9216 C3.299,13.9216 2.991,13.8906 2.69,13.8336 C3.317,15.7896 5.135,17.2136 7.29,17.2536 C5.604,18.5736 3.481,19.3606 1.175,19.3606 C0.777,19.3606 0.385,19.3376 0,19.2926 C2.179,20.6886 4.767,21.5046 7.548,21.5046 C16.605,21.5046 21.557,14.0016 21.557,7.4946 C21.557,7.2816 21.552,7.0696 21.543,6.8586 C22.505,6.1636 23.34,5.2966 24,4.3086"
              fill="#1DA1F2"
              fillRule="evenodd"
            />
          </svg>
        </a>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 fcyNTM"
        />
        <a
          aria-label="Share feedback on Github"
          className="StyledAnchor-sc-1rp7lwl-0 hdTGql"
          href="https://www.github.com/"
          onBlur={[Function]}
          onFocus={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-label="Github"
            className="StyledIcon-ofa7kd-0 fDZBFh"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.9989871,1 C5.92550416,1 1,5.92482888 1,12.0003376 C1,16.8603395 4.15153934,20.9829338 8.52263728,22.4374904 C9.0729918,22.5387827 9.27355045,22.199116 9.27355045,21.9073943 C9.27355045,21.6467356 9.2640965,20.954572 9.25869425,20.0368642 C6.19899322,20.7013414 5.55342398,18.5620492 5.55342398,18.5620492 C5.0530403,17.2911692 4.33183953,16.9528531 4.33183953,16.9528531 C3.33309801,16.2708186 4.40747107,16.2843243 4.40747107,16.2843243 C5.51155652,16.3619816 6.09229872,17.4181221 6.09229872,17.4181221 C7.07348292,19.0988981 8.66714755,18.6133706 9.2938089,18.3317781 C9.39375058,17.6213819 9.67804414,17.1365297 9.99205009,16.86169 C7.54955646,16.5841493 4.98146045,15.6401056 4.98146045,11.4249977 C4.98146045,10.224347 5.41026428,9.24181221 6.11390773,8.47334172 C6.00046042,8.19512569 5.62297799,7.07618404 6.22195279,5.56220265 C6.22195279,5.56220265 7.14506277,5.26642929 9.24653918,6.68992296 C10.12373,6.44547101 11.0650726,6.32392032 12.0003376,6.31919335 C12.9349274,6.32392032 13.8755947,6.44547101 14.7541361,6.68992296 C16.8542619,5.26642929 17.7760214,5.56220265 17.7760214,5.56220265 C18.3763467,7.07618404 17.9988643,8.19512569 17.8860923,8.47334172 C18.5910863,9.24181221 19.0165137,10.224347 19.0165137,11.4249977 C19.0165137,15.6509101 16.444366,16.5807729 13.9944443,16.8529114 C14.3888087,17.192578 14.7406305,17.863808 14.7406305,18.890236 C14.7406305,20.3603241 14.7271248,21.5467939 14.7271248,21.9073943 C14.7271248,22.2018171 14.9256576,22.5441849 15.4834403,22.4368151 C19.8511618,20.9788821 23,16.8589889 23,12.0003376 C23,5.92482888 18.0744958,1 11.9989871,1"
              fill="#333"
              fillRule="evenodd"
            />
          </svg>
        </a>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 fcyNTM"
        />
        <a
          aria-label="Chat with us on Slack"
          className="StyledAnchor-sc-1rp7lwl-0 hdTGql"
          href="https://www.slack.com/"
          onBlur={[Function]}
          onFocus={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-label="Slack"
            className="StyledIcon-ofa7kd-0 fDZBFh"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              fillRule="evenodd"
            >
              <path
                d="M5.04765714 15.1238095C5.04765714 16.5142857 3.92384762 17.6380952 2.53337143 17.6380952 1.14289524 17.6380952.0190857143 16.5142857.0190857143 15.1238095.0190857143 13.7333333 1.14289524 12.6095238 2.53337143 12.6095238L5.04765714 12.6095238 5.04765714 15.1238095zM6.30472381 15.1238095C6.30472381 13.7333333 7.42853333 12.6095238 8.81900952 12.6095238 10.2094857 12.6095238 11.3332952 13.7333333 11.3332952 15.1238095L11.3332952 21.4095238C11.3332952 22.8 10.2094857 23.9238095 8.81900952 23.9238095 7.42853333 23.9238095 6.30472381 22.8 6.30472381 21.4095238L6.30472381 15.1238095z"
                fill="#E01E5A"
              />
              <path
                d="M8.81904762 5.02857143C7.42857143 5.02857143 6.3047619 3.9047619 6.3047619 2.51428571 6.3047619 1.12380952 7.42857143 0 8.81904762 0 10.2095238 0 11.3333333 1.12380952 11.3333333 2.51428571L11.3333333 5.02857143 8.81904762 5.02857143zM8.81904762 6.3048C10.2095238 6.3048 11.3333333 7.42860952 11.3333333 8.81908571 11.3333333 10.2095619 10.2095238 11.3333714 8.81904762 11.3333714L2.51428571 11.3333714C1.12380952 11.3333714 0 10.2095619 0 8.81908571 0 7.42860952 1.12380952 6.3048 2.51428571 6.3048L8.81904762 6.3048z"
                fill="#36C5F0"
              />
              <path
                d="M18.895219 8.81902857C18.895219 7.42855238 20.0190286 6.30474286 21.4095048 6.30474286 22.799981 6.30474286 23.9237905 7.42855238 23.9237905 8.81902857 23.9237905 10.2095048 22.799981 11.3333143 21.4095048 11.3333143L18.895219 11.3333143 18.895219 8.81902857zM17.6380571 8.81902857C17.6380571 10.2095048 16.5142476 11.3333143 15.1237714 11.3333143 13.7332952 11.3333143 12.6094857 10.2095048 12.6094857 8.81902857L12.6094857 2.51426667C12.6094857 1.12379048 13.7332952-.0000190476191 15.1237714-.0000190476191 16.5142476-.0000190476191 17.6380571 1.12379048 17.6380571 2.51426667L17.6380571 8.81902857z"
                fill="#2EB67D"
              />
              <path
                d="M15.1238286 18.8952C16.5143048 18.8952 17.6381143 20.0190095 17.6381143 21.4094857 17.6381143 22.7999619 16.5143048 23.9237714 15.1238286 23.9237714 13.7333524 23.9237714 12.6095429 22.7999619 12.6095429 21.4094857L12.6095429 18.8952 15.1238286 18.8952zM15.1238286 17.6381333C13.7333524 17.6381333 12.6095429 16.5143238 12.6095429 15.1238476 12.6095429 13.7333714 13.7333524 12.6095619 15.1238286 12.6095619L21.4285905 12.6095619C22.8190667 12.6095619 23.9428762 13.7333714 23.9428762 15.1238476 23.9428762 16.5143238 22.8190667 17.6381333 21.4285905 17.6381333L15.1238286 17.6381333z"
                fill="#ECB22E"
              />
            </g>
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
